### PR TITLE
Do not collect docker container logs if they are matching `DD_CONTAINER_EXCLUDE_LOGS`

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Make the agent not collect logs on docker containers that are matching `DD_CONTAINER_EXCLUDE_LOGS` even if they were started before the agent.

### Motivation

Whereas `DD_CONTAINER_EXCLUDE_LOGS` was properly honored on Kubernetes, on docker, it was not working for containers which were started before the agent.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Launch the agent inside docker with `DD_CONTAINER_EXCLUDE_LOGS` set:
```
docker run --rm --name datadog-agent -e DD_API_KEY=$DD_API_KEY -e DD_LOGS_ENABLED=true -e DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL=true -e 'DD_CONTAINER_EXCLUDE_LOGS=image:.*' -v /var/run/docker.sock:/var/run/docker.sock:ro -v /proc/:/host/proc/:ro -v /opt/datadog-agent/run:/opt/datadog-agent/run:rw -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro datadog/agent:latest
```

Without this PR, the agent container logs are collected:
```
docker exec datadog-agent agent status
```
```
==========
Logs Agent
==========
    Sending compressed logs in HTTPS to agent-http-intake.logs.datadoghq.com on port 443
    BytesSent: 9407
    EncodedBytesSent: 850
    LogsProcessed: 23
    LogsSent: 20

  container_collect_all
  ---------------------
    Type: docker
    Status: OK
    Inputs: 07f1a312d5d41d8f064726ab12c8236daca9229a0e0772aa70040fb0f3b55512
```

With this PR, the agent container logs are not collected:
```
docker exec datadog-agent agent status
```
```
==========
Logs Agent
==========

    Sending compressed logs in HTTPS to agent-http-intake.logs.datadoghq.com on port 443
    BytesSent: 0
    EncodedBytesSent: 28
    LogsProcessed: 0
    LogsSent: 0

  container_collect_all
  ---------------------
    Type: docker
    Status: Pending
```

If `DD_CONTAINER_EXCLUDE_LOGS` is changed, logs are collected again.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]
nice catch! would be even nicer if we could unit test the init method.
LGTM otherwise 💯

File: pkg/autodiscovery/listeners/docker.go [hunk 1]
```diff
@@ -148,6 +148,21 @@ func (l *DockerListener) init() {
 			log.Errorf("Error getting check names from docker labels on container %s: %v", co.ID, err)
 		}
 
+		containerImage, err := l.dockerUtil.ResolveImageName(co.Image)
```
Comment (ahmed-mez) [inline pkg/autodiscovery/listeners/docker.go:151]: need to check for error here

Comment (L3n41c) [reply pkg/autodiscovery/listeners/docker.go:151]: Error checked!

File: pkg/autodiscovery/listeners/docker.go [hunk 2]
```diff
@@ -148,6 +148,25 @@ func (l *DockerListener) init() {
...
+		if err != nil {
+			log.Warnf("error while resolving image name: %s", err)
```
Comment (ahmed-mez) [inline pkg/autodiscovery/listeners/docker.go:153]: nit
```suggestion
			log.Warnf("Error while resolving image name: %s", err)
```

File: releasenotes/notes/honor-DD_CONTAINER_EXCLUDE_LOGS-f42c6b2fa8aa6b32.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    The agent was collecting docker container logs even if they are matching `DD_CONTAINER_EXCLUDE_LOGS`
```
Comment (ahmed-mez) [inline releasenotes/notes/honor-DD_CONTAINER_EXCLUDE_LOGS-f42c6b2fa8aa6b32.yaml:11]: should we mention `DD_CONTAINER_EXCLUDE_METRICS` as well?

File: pkg/autodiscovery/listeners/docker.go (hunk 2 above)
Comment (L3n41c) [reply pkg/autodiscovery/listeners/docker.go]: Fixed!

File: releasenotes/notes/honor-DD_CONTAINER_EXCLUDE_LOGS-f42c6b2fa8aa6b32.yaml (hunk 3 above)
Comment (L3n41c) [reply releasenotes/notes/honor-DD_CONTAINER_EXCLUDE_LOGS-f42c6b2fa8aa6b32.yaml]: Indeed. I’ve added `DD_CONTAINER_EXCLUDE_METRICS`.

**Pull request merged**